cmd/environment_clone.go

```diff
@@ -34,56 +30,9 @@ var environmentCloneCmd = &cobra.Command{
 			panic("unreachable") // staticcheck false positive: https://staticcheck.io/docs/checks#SA5011
 		}
 
-		req := qovery.CloneEnvironmentRequest{
-			Name:                newEnvironmentName,
-			ApplyDeploymentRule: &applyDeploymentRule,
-		}
-
-		if clusterName != "" {
-			clusters, _, err := client.ClustersAPI.ListOrganizationCluster(context.Background(), orgId).Execute()
-
-			if err != nil {
-				utils.PrintlnError(err)
-				os.Exit(1)
-				panic("unreachable") // staticcheck false positive: https://staticcheck.io/docs/checks#SA5011
-			}
-
-			if err == nil {
-				for _, c := range clusters.GetResults() {
-					if strings.EqualFold(c.Name, clusterName) {
-						req.ClusterId = &c.Id
-						break
-					}
-				}
-			}
-		}
-
-		if environmentType != "" {
-			switch strings.ToUpper(environmentType) {
-			case "DEVELOPMENT":
-				req.Mode = qovery.EnvironmentModeEnum.Ptr(qovery.ENVIRONMENTMODEENUM_DEVELOPMENT)
-			case "PRODUCTION":
-				req.Mode = qovery.EnvironmentModeEnum.Ptr(qovery.ENVIRONMENTMODEENUM_PRODUCTION)
-			case "STAGING":
-				req.Mode = qovery.EnvironmentModeEnum.Ptr(qovery.ENVIRONMENTMODEENUM_STAGING)
-			}
-		}
-
-		_, res, err := client.EnvironmentActionsAPI.CloneEnvironment(context.Background(), envId).CloneEnvironmentRequest(req).Execute()
-
-		if err != nil {
-			// print http body error message
-			if res != nil  && !strings.Contains(res.Status, "200") {
-				result, _ := io.ReadAll(res.Body)
-				utils.PrintlnError(errors.Errorf("status code: %s ; body: %s", res.Status, string(result)))
-			}
-
-			utils.PrintlnError(err)
-			os.Exit(1)
-			panic("unreachable") // staticcheck false positive: https://staticcheck.io/docs/checks#SA5011
-		}
+		var newEnvId = environment.EnvironmentClone(client, organizationName, projectName, environmentName, newEnvironmentName, clusterName, environmentType, applyDeploymentRule, orgId, envId)
 
-		utils.Println("Environment is cloned!")
+		utils.Println("your new environment ID is: " + newEnvId.Id)
```

Comment:
Maybe we can improve the message, e.g:
```suggestion
		utils.Println(fmt.Sprintf("Your environment has been cloned [id: %s, name: %s]", newEnv.Id, newEnv.name))
```